assignments/Assignment4.js

```diff
@@ -0,0 +1,48 @@
+
+// #2 Return the largest of 3 arguments numeric or string p.112
+function max3(x, y, z) {
+    if ((x > y) && (x > z)) {
+        return console.log(x + " is the largest");
```

Comment:
You should be returning the largest value itself.  This code currently returns the result of calling `console.log` which is undefined (i.e. `console.log` doesn't itself return a value).

Reply:
It is working when I run it in the browser. I was googling and came across this but from your lecture tonight your saying console.log should not be returned for results. So I should have returned the value then console.log to print.
function max3(x, y, z) {
    if ((x > y) && (x > z)) {
        return console.log(x + " is the largest");
    } else if ((y > x) && (y > z)) {
        return console.log(y + " is the largest");
    } else {
        return console.log(z + " is the largest");
    }
}

max3(30, 50, 20)
max3("bite", "bat", "bit")
max3(3.14, 3.141, -3.145)

50 is the largest
bite is the largest
3.141 is the largest

Reply:
// #2 Return the largest of 3 arguments numeric or string p.112
function max3(x, y, z) {
    if ((x > y) && (x > z)) {
        return x;
    } else if ((y > x) && (y > z)) {
        return y;
    } else {
        return z;
    }
}
console.log(max3(30, 50, 20));
console.log(max3("bite", "bat", "bit"));
console.log(max3(3.14, 3.141, -3.145));